core/drivers/regulator/regulator.c

```diff
@@ -323,9 +303,187 @@ static TEE_Result regulator_core_cleanup(void)
 		}
 	}
 
-	regulator_print_state(__func__);
+	if (TRACE_LEVEL >= TRACE_DEBUG)
```

Comment:
>= TRACE_INFO?

Reply:
The previous implementation printed the regulator tree with TRACE_DEBUG level so I've replicated the behavior. Makes sense?

Reply:
I think the regulator tree should not be part of the very few info level traces printed at OP-TEE boot-time when TRACE_LEVEL is TRACE_INFO.
I could address your concern replacing use of `IMSG()` with `MSG()` in `regulator_print_tree()` and related functions.

Reply:
Then why not use DMSG() if it's a debug message?

Reply:
`DMSG` would require TRACE_DEBUG level to be enabled, which is too often too verbose. Using TRACE_INFO level makes OP-TEE far less verbose, still aloowing one to print some info (as regu/clock trees) when needed.
That's not a strong opinion but I think it's more handy.

Reply:
> `DMSG` would require TRACE_DEBUG level to be enabled, which is too often too verbose

But `if (TRACE_LEVEL >= TRACE_DEBUG)` also requires DEBUG to be enabled! (or FLOW which is even more verbose).
The code as it is proposed does not print anything when TRACE_LEVEL is TRACE_ERROR or TRACE_INFO, and it prints something (with severity INFO) when TRACE_LEVEL is TRACE_DEBUG or TRACE_FLOW in which case all DEBUG messages are always there too. So I don't understand.

This is debugging information. IMO the right thing is just to use `DMSG()` (or `DMSG_RAW()` if you'd rather not have the file/line number which makes sense) and no `if`. It's not like we need to optimize anything here.

Reply:
Ok, let's make it simple. I'll switch to `DMSG()`.

Reply:
... and update the commit message, _mk/config.mk_ and _regulator.h_ inline description comments accordingly.